Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -405,6 +405,8 @@ void ImageEditor::resize_event(GUI::ResizeEvent& event)
 
 void ImageEditor::keydown_event(GUI::KeyEvent& event)
 {
+    if (event.key() == Key_Delete && !selection().is_empty() && active_layer())
+        selection().erase_selection(active_layer()->bitmap());
```

Comment:
Since you're consuming the event, perhaps you should return here and not pass it along to the active tool?